ambry-named-mysql/src/test/java/com/github/ambry/named/MySqlNamedBlobDbTest.java

```diff
@@ -141,6 +145,25 @@ public void testPullAndCleanStaleNamedBlobs() throws Exception {
     namedBlobDb.cleanupStaleData(staleNamedBlobs);
   }
 
+  @Test
+  public void testUpdateBlobStateToReady() throws Exception {
+    Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    calendar.add(Calendar.DATE, 10);
+
+    dataSourceFactory.setLocalDatacenter(foundDatacenter);
+    dataSourceFactory.triggerDataResultSet(datacenters);
+
+    long expirationTimeMs = calendar.getTimeInMillis() + 5000;
+    final NamedBlobRecord record = new NamedBlobRecord(account.getName(), container.getName(), "blobName", id, expirationTimeMs);
+    PutResult putResult = namedBlobDb.put(record, NamedBlobState.IN_PROGRESS, true).get();
+
+    record.setVersion(putResult.getInsertedRecord().getVersion());
+    namedBlobDb.updateBlobStateToReady(record).get();
+
+    NamedBlobRecord namedBlobRecord = namedBlobDb.get(account.getName(), container.getName(), "blobName").get();
+    assertEquals("Blob Id is not matched with the record", id, namedBlobRecord.getBlobId());
```

Comment:
we should check that before the update we can't get anything out, and then with the update, we can get the record out.

Reply:
I just added the check in the [integration test](https://github.com/linkedin/ambry/pull/2383/files#diff-c568b3a93a06b59db49d461110e676415b73b4dfb1b7c9035afd1929e61bacc3R531).
Here it's using mock all the places in unit test